src/dns.rs

```diff
@@ -88,3 +90,28 @@ impl ToSocketAddrs for (IpAddr, u16) {
         (*self).into()
     }
 }
+
+impl<T: ToSocketAddrs + ?Sized> ToSocketAddrs for &T {
+    fn to_socket_addr(&self, dns: &Dns) -> SocketAddr {
+        (**self).to_socket_addr(dns)
+    }
+}
+
+impl ToSocketAddrs for str {
+    fn to_socket_addr(&self, _: &Dns) -> SocketAddr {
+        self.parse().unwrap()
+    }
+}
+
+impl ToSocketAddrs for String {
+    fn to_socket_addr(&self, _: &Dns) -> SocketAddr {
+        self.parse().unwrap()
+    }
+}
```

Comment:
shouldn't we be doing some DNS lookups here like in https://github.com/tokio-rs/turmoil/blob/main/src/dns.rs#L65-L90?

Reply:
oh yes I think I copied the tokio code oops

Reply:
wait no some of them don't go through dns and some do. We will need to fix that.